src/netutil.rs

```diff
@@ -1,15 +1,28 @@
 //! The `netutil` module assists with networking
```

Comment:
would prefer this stuff ended up in its own file, is probably enough to make a crate of its own...